### PR TITLE
Fix #3141: Avoid infinite reload of the page when offline

### LINKED ISSUE #3141: mobile.html redirect to mobile.htmlmobile.html recursively.
This happens when there is no or bad  internet access (only mobile version)

### PR DESCRIPTION
**Pull request by oterral**
[Branch](https://mf-geoadmin3.dev.bgdi.ch/teo_fixes/)

Fix #3141

### CODE REVIEW

**Comment by loicgasser** [comment]
is there a related issue ?

**Comment by loicgasser** [comment]
Oh yeah in title

**Comment by loicgasser** [comment]
Why getting rid of the closure ?

**Comment by oterral** [comment]
Because it's useless.

**Comment by oterral** [comment]
Arf yes that's add some useless variables  to the window object. I can re add it.

**Comment by loicgasser** [comment]
I don't want to be pedantic but it s really hard to read now

**Comment by oterral** [comment]
I know but this part is not compressed in the final index.html.

File: src/index.mako.html [hunk 1]
```diff
@@ -70,37 +70,24 @@
...
-          redirectToMobile();
+        m = (m) ? m[1] : m; // avoid loops
+        if (!(p == 'desktop' && m == 'false') && !(p == 'mobile' && m == 'true')) {
+          var href, onMobile = (('ontouchstart' in w) || n.maxTouchPoints > 1 || n.msMaxTouchPoints > 1) && (screen.width <= 768 || screen.height <= 768);
+          if (m == 'false' || (!onMobile && p == 'mobile')) {
+            href = l.origin + l.pathname.replace(/\/(|(mobile|embed)\.html)$/, '/index.html') + l.search;
+          } else if (m == 'true' || (onMobile && p == 'desktop')) {
+            href = l.origin + l.pathname.replace(/\/(|(index|embed)\.html)$/, '/mobile.html') + l.search;
+          }
+          if (href && href!=l.href) {
```
Comment (loicgasser) [inline src/index.mako.html]: I see the fix now

File: src/index.mako.html [hunk 2]
```diff
@@ -70,37 +70,24 @@
...
+            l.href = href;
+          }
         }
-      })();
+      })()
```
Comment (loicgasser) [inline src/index.mako.html]: ;

File: src/index.mako.html (hunk 1 above)
Comment (gjn) [inline src/index.mako.html]: spaces between parameters and operators.

File: src/index.mako.html [hunk 3]
```diff
@@ -70,37 +70,24 @@
...
-
-        var redirectToMobile = function() {
-          pathname = !/(index|embed)\.html$/.test(l.pathname) ?
-            l.pathname + 'mobile.html' : l.pathname.replace(/(index|embed)\.html$/, 'mobile.html');
-          l.href = l.origin + pathname + l.search;
-        }
-        var redirectToDesktop = function() {
-          l.href = l.origin + l.pathname.replace(/(mobile|embed)\.html$/, 'index.html') + l.search;
+      (function() {
+        var w=window, l=w.location, n=w.navigator, p='${device}', m=l.search.match(/(?:mobile=(true|false))/);
```
Comment (gjn) [inline src/index.mako.html]: spaces...

File: src/index.mako.html [hunk 4]
```diff
@@ -70,37 +70,24 @@
...
+        if (!l.origin) {
+          l.origin=l.protocol + "//" +l.hostname; // IE fix
```
Comment (gjn) [inline src/index.mako.html]: spaces.

**Comment by gjn** [comment]
Some formatting issues. Otherwise, LGTM.

Please merge when adapted.

**Comment by loicgasser** [comment]
Thank you for adaptation LGTM

**Comment by oterral** [comment]
Thx for the reviews!!!

**Pull request merged**